src/workerd/api/r2-rpc.h

```diff
@@ -82,6 +83,7 @@ kj::Promise<R2Result> doR2HTTPPutRequest(
     kj::Maybe<uint64_t> streamSize,
     // Deprecated. For internal beta API only.
     kj::String metadataPayload,
-    kj::Maybe<kj::String> path);
+    kj::Array<kj::String> path,
```

Comment:
I think I'm missing context as to why this needs to become an array. Why not just store the path in a `kj::String` in R2Bucket?

Reply:
Has to do with how `kj::URL.add()` behaves. It expects a separate call to the `path.add()` for each slash-separated piece of the URL. https://github.com/capnproto/capnproto/blob/master/c++/src/kj/compat/url.h

At least that's what was observed while testing. Without this, `/` would be URL-encoded to `%2F`.

Reply:
Ah ok. Unless @jasnell / @kentonv have an alternate suggestion, then yeah, seems like we should be using an array across the board now to inject things properly.

Reply:
That being said, I feel like these should be passing in a `kj::ArrayPtr` instead of `kj::Array` to avoid a heap allocation. The RPC call doesn't need to take over ownership of the array you pass in.